src/esphome/sensor/ppd42x_sensor.h

```diff
@@ -0,0 +1,59 @@
+#ifndef ESPHOME_SENSOR_PPD42X_SENSOR_H
+#define ESPHOME_SENSOR_PPD42X_SENSOR_H
+
+#include "esphome/defines.h"
+
+#ifdef USE_PPD42X_SENSOR
+
+#include "esphome/sensor/sensor.h"
+#include "esphome/esphal.h"
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace sensor {
+
+class Ppd42xSensorComponent : public PollingSensorComponent {
```

Comment:
This class shouldn't inherit from `PollingSensorComponent` but from `PollingComponent`. In your current design both PM10 and pm2.5 values will be published to the same sensor, which is wrong. Please see PMSx003 example for how to split the two values.